apps/heft/src/utilities/TaskPackageResolver.ts

```diff
@@ -0,0 +1,254 @@
+    terminal: Terminal
+  ): string | undefined {
+    if (visitedTsconfigPaths.has(tsconfigPath)) {
+      throw new Error(`The file "${tsconfigPath}" has an "extends" field that creates a circular reference`);
+    }
+    visitedTsconfigPaths.add(tsconfigPath);
+
+    terminal.writeVerboseLine(`Examining ${tsconfigPath}`);
+
+    if (!FileSystem.exists(tsconfigPath)) {
+      throw new Error(`The referenced tsconfig.json file does not exist: ` + tsconfigPath);
+    }
+
+    let tsconfig: ITsconfig;
+    try {
+      tsconfig = JsonFile.load(tsconfigPath);
+    } catch (e) {
+      throw new Error(`Error parsing tsconfig.json: ${e}\n` + tsconfigPath);
+    }
+
+    if (tsconfig.extends) {
+      // Follow the tsconfig.extends field:
+      let baseTsconfigPath: string;
+      if (path.isAbsolute(tsconfig.extends)) {
+        // Absolute path
+        terminal.writeVerboseLine(
+          `Following a tsconfig.json "extends" property "${tsconfig.extends}" that is an absolute path.`
+        );
+        baseTsconfigPath = tsconfig.extends;
+      } else if (tsconfig.extends.match(/^\./)) {
+        // Relative path
+        terminal.writeVerboseLine(
+          `Following a tsconfig.json "extends" property "${tsconfig.extends}" that is a relative path.`
+        );
+        baseTsconfigPath = path.resolve(path.dirname(tsconfigPath), tsconfig.extends);
+      } else {
+        terminal.writeVerboseLine(
+          `Following a tsconfig.json "extends" property "${tsconfig.extends}" that is a package path.`
+        );
+        // Package path
+        baseTsconfigPath = resolve.sync(tsconfig.extends, {
+          basedir: path.dirname(tsconfigPath),
+          preserveSymlinks: false,
+          packageFilter: (packageJson: INodePackageJson) => {
+            return {
+              ...packageJson,
+              // ensure "main" points to a file in the package root folder
+              main: 'package.json'
+            };
+          }
+        });
+      }
+
+      terminal.writeVerboseLine(`Resolved "extends" path to: ${baseTsconfigPath}`);
+      const result: string | undefined = TaskPackageResolver._locateRigPackageFolder(
+        baseTsconfigPath,
+        visitedTsconfigPaths,
+        visitedRigPackagePaths,
+        terminal
+      );
+      if (result) {
+        // We found the rig via baseTsconfigPath, so we're done
+        return result;
+      }
+    }
+
+    // Look for the governing package of "tsconfigPath"
+    const rigPackagePath: string | undefined = TaskPackageResolver._packageJsonLookup.tryGetPackageFolderFor(
+      tsconfigPath
+    );
+    if (!rigPackagePath) {
+      // This is unexpected; for now, we'll treat it as unsupported.  Please open a GitHub issue if you find
+      // a legitimate reason to reference tsconfig.json that is not part of some package.
+      throw new Error(
+        'No associated package.json was found for the tsconfig.json referenced via "extends":' + tsconfigPath
```

Comment:
May be a bit confusing, given that the dev may not know where this tsconfig path came from (since it doesn't tell you the source file that has this "extends" prop value). May not be a big deal, just wanted to point it out

Reply:
I actually couldn't get this error to occur, because if the package.json file is missing, then other errors problems occur earlier. I'm not going to worry about this right now -- better to see what problems people encounter in real life and then tune the messages around that.